components/canvas/gl_context.rs

```diff
@@ -168,44 +193,101 @@ impl GLContextWrapper {
             GLContextWrapper::OSMesa(ref ctx) => {
                 WebGLImpl::apply(ctx, state, cmd, backtrace);
             },
+            #[cfg(target_os = "macos")]
+            GLContextWrapper::NativeWithIOSurface(ref ctx) => {
+                WebGLImpl::apply(ctx, state, cmd, backtrace);
+            },
         }
     }
 
     pub fn gl(&self) -> &dyn gl::Gl {
         match *self {
             GLContextWrapper::Native(ref ctx) => ctx.gl(),
             GLContextWrapper::OSMesa(ref ctx) => ctx.gl(),
+            #[cfg(target_os = "macos")]
+            GLContextWrapper::NativeWithIOSurface(ref ctx) => ctx.gl(),
         }
     }
 
-    pub fn get_info(&self) -> (Size2D<i32>, u32, GLLimits) {
+    pub fn get_info(&self) -> (Size2D<i32>, u32, Option<u32>, GLLimits) {
```

Comment:
I think we can have get_info return `(Size2D<i32>, TextureBacking, GLLimits)` instead.